src/token_map_impl.hpp

```diff
@@ -476,15 +487,17 @@ void ReplicationStrategy<Partitioner>::build_replicas_network_topology(
       // datacenter only then consider hosts in the same rack
 
       if (rack == 0 || racks_observed_this_dc.size() == rack_count_this_dc) {
-        if (add_replica(replicas, Host::Ptr(host))) {
+        if (replicas_set.insert(host).second) {
+          replicas->push_back(Host::Ptr(host));
           ++replica_count_this_dc;
         }
       } else {
         TokenHostQueue& skipped_endpoints_this_dc = dc_rack_info.skipped_endpoints;
         if (racks_observed_this_dc.count(rack) > 0) {
           skipped_endpoints_this_dc.push_back(curr_token_it);
         } else {
-          if (add_replica(replicas, Host::Ptr(host))) {
+          if (replicas_set.insert(host).second) {
```

Comment:
why not just embed the changes in the `add_replica()`?
another option is to try keep `replicas` sorted and use binary search? i.e. use 
`std::sort()`, or even `std::stable_sort()` that should be faster for sorted or, almost sorted data, then just use `std::lower_bound()`  with custom `Comparator` to check if the host needs to be added.

Reply:
The token order of `replicas` matters.

Reply:
still could embed the logic in to `add_replica()`